plugins/azure/fix_plugin_azure/azure_client.py

```diff
@@ -128,16 +156,45 @@ def _update_or_delete_tag(self, tag_name: str, tag_value: str, resource_id: str,
             updated_resource = GenericResource(location=resource.location, tags=resource.tags)
             self.client.resources.begin_create_or_update_by_id(resource_id, "2021-04-01", updated_resource)
 
+        except ResourceNotFoundError:
+            return False  # Resource not found
+        except ResourceExistsError:
+            return False  # Tag for update/delete does not exist
         except HttpResponseError as e:
-            if e.error and e.error.code == "ResourceNotFoundError":
-                return False  # Resource not found
-            elif e.error and e.error.code == "ResourceExistsError":
-                return False  # Tag for update/delete does not exist
-            else:
-                raise e
+            raise e
+        except Exception as e:
+            raise e
 
         return True
 
+    @retry(  # type: ignore
+        stop_max_attempt_number=3,  # max 3 attempts
```

Comment:
There are 10 in AWS, and I would also use it here.
Fix is trying hard to get the data - otherwise, we would see an incomplete state.